modules/kubernetes-addons/aws-ebs-csi-driver/main.tf

```diff
@@ -1,10 +1,13 @@
 locals {
   create_irsa = try(var.addon_config.service_account_role_arn == "", true)
+  name        = try(var.helm_config.name, "aws-ebs-csi-driver")
+  namespace   = try(var.helm_config.namespace, "kube-system")
 }
 
 resource "aws_eks_addon" "aws_ebs_csi_driver" {
+  count                    = var.enable_amazon_eks_aws_ebs_csi_driver && !var.enable_self_managed_aws_ebs_csi_driver ? 1 : 0
   cluster_name             = var.addon_context.eks_cluster_id
-  addon_name               = "aws-ebs-csi-driver"
```

Comment:
this must stay as is - the name is not optional/configurable when using EKS addon